### PR TITLE
f4pga: toolchain wrappers (shell) moved from arch-defs

### PR DESCRIPTION
**Pull request by umarcor**

### CODE REVIEW

File: f4pga/setup.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+wrapper_entrypoints = [
+    f"{sf}_generate_constraints = {shwrappers}:generate_constraints",
+    f"{sf}_pack = {shwrappers}:pack",
+    f"{sf}_place = {shwrappers}:place",
+    f"{sf}_route = {shwrappers}:route",
+    f"{sf}_synth = {shwrappers}:synth",
+    f"{sf}_write_bitstream = {shwrappers}:write_bitstream",
+    f"{sf}_write_fasm = {shwrappers}:write_fasm",
+] if F4PGA_FAM == 'xc7' else [
+#    f"{sf}_generate_constraints = {shwrappers}:generate_constraints",
```
Comment (kgugala) [inline f4pga/setup.py:43]: please remove commented out code

File: f4pga/wrappers/sh/__init__.py [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+
+
+def run_sh(script):
+    stdout.flush()
+    stderr.flush()
+    check_call([str(script)]+sys_argv[1:])
+
+
+def generate_constraints():
+    print("F4PGA (old) generate constraints")
```
Comment (kgugala) [inline f4pga/wrappers/sh/__init__.py:39]: we should have better message here ("old" doesn't mean much for new users). Maybe we should call it "deprecated"?

File: f4pga/wrappers/sh/quicklogic/conda_build_install_package.sh [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+wget https://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh -O conda_installer.sh
+bash conda_installer.sh -b -p $INSTALL_DIR/conda && rm conda_installer.sh
+source "$INSTALL_DIR/conda/etc/profile.d/conda.sh"
+echo "include-system-site-packages=false" >> $INSTALL_DIR/conda/pyvenv.cfg
+CONDA_FLAGS="-y --override-channels -c defaults -c conda-forge"
+conda update $CONDA_FLAGS -q conda
+curl $(curl https://storage.googleapis.com/symbiflow-arch-defs-install/latest-qlf) > arch.tar.gz
+tar -C $INSTALL_DIR -xvf arch.tar.gz && rm arch.tar.gz
+conda install $CONDA_FLAGS -c litex-hub/label/main yosys="0.9_5266_g0fb4224e 20210301_104249_py37"
```
Comment (kgugala) [inline f4pga/wrappers/sh/quicklogic/conda_build_install_package.sh:44]: why not use env.yml file?

With the code we have now we'll need to change this script on every update

**Review by kgugala — CHANGES_REQUESTED** [review]

File: f4pga/setup.py (hunk 1 above)
Comment (umarcor) [reply f4pga/setup.py]: Done.

File: f4pga/wrappers/sh/__init__.py (hunk 2 above)
Comment (umarcor) [reply f4pga/wrappers/sh/__init__.py]: Done.

File: f4pga/wrappers/sh/quicklogic/conda_build_install_package.sh (hunk 3 above)
Comment (umarcor) [reply f4pga/wrappers/sh/quicklogic/conda_build_install_package.sh:44]: This is copied as-is from https://github.com/SymbiFlow/f4pga-arch-defs/blob/main/quicklogic/common/toolchain_wrappers/conda_build_install_package.sh. Enhancements related to QL wrappers are expected to be pushed to this repo after merging this PR.

Comment (kgugala) [reply f4pga/wrappers/sh/quicklogic/conda_build_install_package.sh:44]: ok

**Review by kgugala — APPROVED** [review]

**Pull request merged**